@vue/eslint-config-typescript is implicitly dependent on eslint-plugin-vue

### Version
3.0.0-rc.3

### Reproduction link
[https://github.com/st-sloth/vue-cli-typescript-eslint-parser](https://github.com/st-sloth/vue-cli-typescript-eslint-parser)

### Steps to reproduce
```
npm i
npm run lint-parserOptions-parser
```

### What is expected?
Probably, no errors should be reported.

### What is actually happening?
```
  1:7  error  Parsing error: Unexpected token :
```

---
The issue is that `eslint-config-typescript` puts `parser` inside of `parserOptions` instead of the root. Most likely because it is silently expected that `eslint-plugin-vue` is also used and [it defines `parser` as `vue-eslint-parser` that in turn uses `parserOptions.parser`, defined in `eslint-config-typescript`](https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error) (with this configuration, there is no error - `npm run lint-vue-plugin` in repro).

In our team we use sharable eslint configs that extend great vue configs and are split by feature - our TypeScript partial config extends `eslint-config-typescript`. And in the project using TypeScript without using Vue we now faced the problem that `typescript-eslint-parser` does not get used as the parser.
Of course, all this is probably by design, and we must explicitly add `typescript-eslint-parser` in our config (as `npm run lint-parser` in repro), but I've only come to this realization after hours of debugging, searching through the docs and code of different related repos and such.
Perhaps, at least the reason why in `eslint-config-typescript` `parser` is defined inside `parserOptions` should be explained in the config and/or docs and readers should be warned that in order to use this config without `eslint-pliugin-vue` they must specify parser themselves?

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/docs/.vuepress/config.js
+++ b/docs/.vuepress/config.js
@@ -11,7 +11,7 @@ module.exports = {
   themeConfig: {
     repo: 'vuejs/vue-cli',
     docsDir: 'docs',
-    docsBranch: 'dev',
+    docsBranch: 'docs',
     editLinks: true,
     sidebarDepth: 3,
     locales: {

--- a/docs/config/README.md
+++ b/docs/config/README.md
@@ -8,7 +8,7 @@ sidebar: auto
 
 ## Global CLI Config
 
-Some global configurations for `@vue/cli`, such as your preferred package manager and your locally saved presets, are stored in a JSON file named `.vuerc` in your home directory. You can edit this file directory with your editor of choice to change the saved options.
+Some global configurations for `@vue/cli`, such as your preferred package manager and your locally saved presets, are stored in a JSON file named `.vuerc` in your home directory. You can edit this file directly with your editor of choice to change the saved options.
 
 You can also use the `vue config` command to inspect or modify the global CLI config.
 

--- a/docs/guide/README.md
+++ b/docs/guide/README.md
@@ -20,7 +20,7 @@ Vue CLI is a full system for rapid Vue.js development, providing:
   - Configurable via in-project config file;
   - Extensible via plugins
 - A rich collection of official plugins integrating the best tools in the frontend ecosystem.
-- A full graphical user intefrace to create and manage Vue.js projects.
+- A full graphical user interface to create and manage Vue.js projects.
 
 Vue CLI aims to be the standard tooling baseline for the Vue ecosystem. It ensures the various build tools work smoothly together with sensible defaults so you can focus on writing your app instead of spending days wrangling with configurations. At the same time, it still offers the flexibility to tweak the config of each tool without the need for ejecting.
 

--- a/docs/guide/html-and-static-assets.md
+++ b/docs/guide/html-and-static-assets.md
@@ -108,9 +108,9 @@ Internally, we use `file-loader` to determine the final file location with versi
 
 ### The `public` Folder
 
-Any static assets placed in the `public` folder will simply be copied and not go through webpack. You need to reference to them using absolute paths.
+Any static assets placed in the `public` folder will simply be copied and not go through webpack. You need to reference them using absolute paths.
 
-Note we recommended importing assets as part of your module dependency graph so that they will go through webpack with the following benefits:
+Note we recommend importing assets as part of your module dependency graph so that they will go through webpack with the following benefits:
 
 - Scripts and stylesheets get minified and bundled together to avoid extra network requests.
 - Missing files cause compilation errors instead of 404 errors for your users.

--- a/docs/guide/installation.md
+++ b/docs/guide/installation.md
@@ -9,7 +9,7 @@ If you have the previous `vue-cli` (1.x or 2.x) package installed globally, you 
 Vue CLI requires [Node.js](https://nodejs.org/) version 8.9 or above (8.11.0+ recommended). You can manage multiple versions of Node on the same machine with [nvm](https://github.com/creationix/nvm) or [nvm-windows](https://github.com/coreybutler/nvm-windows).
 :::
 
-To install the new package, use on of those commands:
+To install the new package, use one of those commands:
 
 ``` bash
 npm install -g @vue/cli

--- a/packages/@vue/eslint-config-airbnb/README.md
+++ b/packages/@vue/eslint-config-airbnb/README.md
@@ -1,3 +1,11 @@
 # @vue/eslint-config-airbnb
 
 > eslint-config-airbnb for vue-cli
+
+This config is specifically designed to be used by `vue-cli` setups
+and is not meant for outside use (it can be used but some adaptations
+on the user side might be needed - for details see the config file).
+
+A part of its design is that this config may implicitly depend on
+other parts of `vue-cli` setups, such as `eslint-plugin-vue` being
+extended in the same resulting config.

--- a/packages/@vue/eslint-config-prettier/README.md
+++ b/packages/@vue/eslint-config-prettier/README.md
@@ -1,3 +1,11 @@
 # @vue/eslint-config-prettier
 
 > eslint-config-prettier for vue-cli
+
+This config is specifically designed to be used by `vue-cli` setups
+and is not meant for outside use (it can be used but some adaptations
+on the user side might be needed - for details see the config file).
+
+A part of its design is that this config may implicitly depend on
+other parts of `vue-cli` setups, such as `eslint-plugin-vue` being
+extended in the same resulting config.

--- a/packages/@vue/eslint-config-standard/README.md
+++ b/packages/@vue/eslint-config-standard/README.md
@@ -1,3 +1,11 @@
 # @vue/eslint-config-standard
 
 > eslint-config-standard for vue-cli
+
+This config is specifically designed to be used by `vue-cli` setups
+and is not meant for outside use (it can be used but some adaptations
+on the user side might be needed - for details see the config file).
+
+A part of its design is that this config may implicitly depend on
+other parts of `vue-cli` setups, such as `eslint-plugin-vue` being
+extended in the same resulting config.

--- a/packages/@vue/eslint-config-typescript/README.md
+++ b/packages/@vue/eslint-config-typescript/README.md
@@ -1,3 +1,11 @@
 # @vue/eslint-config-typescript
 
 > eslint-config-typescript for vue-cli
+
+This config is specifically designed to be used by `vue-cli` setups
+and is not meant for outside use (it can be used but some adaptations
+on the user side might be needed - for details see the config file).
+
+A part of its design is that this config may implicitly depend on
+other parts of `vue-cli` setups, such as `eslint-plugin-vue` being
+extended in the same resulting config.

--- a/packages/@vue/eslint-config-typescript/index.js
+++ b/packages/@vue/eslint-config-typescript/index.js
@@ -1,5 +1,9 @@
 module.exports = {
   plugins: ['typescript'],
+  // Prerequisite `eslint-plugin-vue`, being extended, sets
+  // root property `parser` to `'vue-eslint-parser'`, which, for code parsing,
+  // in turn delegates to the parser, specified in `parserOptions.parser`:
+  // https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error
   parserOptions: {
     parser: require.resolve('typescript-eslint-parser')
   },

